### PR TITLE
adding -outpubkey option to genpkey

### LINKED ISSUE #22132: Add -pubout option to genpkey to output the public key
This is needed for algorithms where the public key is not contained within and cannot be generated from the private key.

### PR DESCRIPTION
**Pull request by baentsch**
Adding the option `-outpubkey <filename>` to output a public key via the [genpkey](https://www.openssl.org/docs/manmaster/man1/openssl-genpkey.html) command.

Fixes #22132

### CODE REVIEW

File: apps/genpkey.c [hunk 1]
```diff
@@ -228,6 +233,12 @@ int genpkey_main(int argc, char **argv)
     if (out == NULL)
         goto end;
 
+    if (outpubkeyfile != NULL) {
+        outpubkey = bio_open_owner(outpubkeyfile, outformat, private);
```
Comment (beldmit) [inline apps/genpkey.c:237]: Should it be freed somewhere?

Comment (baentsch) [reply apps/genpkey.c:237]: *blush*. Done.

**Review by beldmit — APPROVED** [review]
LGTM if tests passed

**Review by t8m — APPROVED** [review]

**Comment by beldmit** [comment]
Squashed and merged. Thanks for the contribution!

**Pull request closed without merging**